segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -0,0 +1,450 @@
+
+    /**
+     * Epoch of the current instance.
+     */
+    @Getter
+    private long epoch;
+
+    /**
+     * Container id of the owner container.
+     */
+    @Getter
+    private int containerId;
+
+    /**
+     * Index of current journal file.
+     */
+    @Getter
+    private int currentFileIndex;
+
+
+    /**
+     * Sting prefix for all system segments.
+     */
+    @Getter
+    @Setter
+    private String systemSegmentsPrefix;
+
+    /**
+     * System segments to track.
+     */
+    @Getter
+    @Setter
+    private String[] systemSegments;
+
+    /**
+     * Offset at which next log will be written.
+     */
+    private long systemJournalOffset;
+
+    /**
+     * Default {@link SegmentRollingPolicy} for the system segments.
+     */
+    @Getter
+    private SegmentRollingPolicy segmentRollingPolicy;
+
+    /**
+     * Constructs an instance of {@link SystemJournal}.
+     *
+     * @param containerId Container id of the owner container.
+     * @param epoch Epoch of the current container instance.
+     * @param chunkStorage ChunkStorageProvider instance to use for writing all logs.
+     * @param metadataStore ChunkMetadataStore for owner container.
+     * @param segmentRollingPolicy Default SegmentRollingPolicy for system segments.
+     * @throws Exception In case of any errors.
+     */
+    public SystemJournal(int containerId, long epoch, ChunkStorageProvider chunkStorage, ChunkMetadataStore metadataStore, SegmentRollingPolicy segmentRollingPolicy) throws Exception {
+        this.chunkStorage = Preconditions.checkNotNull(chunkStorage, "chunkStorage");
+        this.metadataStore = Preconditions.checkNotNull(metadataStore, "metadataStore");
+        this.segmentRollingPolicy = Preconditions.checkNotNull(segmentRollingPolicy, "segmentRollingPolicy");
+        this.containerId = containerId;
+        this.epoch = epoch;
+        this.systemSegments = getChunkStorageSystemSegments(containerId);
+        this.systemSegmentsPrefix = NameUtils.INTERNAL_SCOPE_NAME;
+
+        Preconditions.checkState(!chunkStorage.exists(getSystemJournalFileName()));
+    }
+
+    /**
+     * Initializes this instance.
+     *
+     * @throws Exception Exception if any.
+     */
+    public void initialize() throws Exception {
+        chunkStorage.create(getSystemJournalFileName());
+    }
+
+    /**
+     * Commits a given system log record to the underlying log chunk.
+     *
+     * @param logLine Log line to persist.
+     * @throws IOException Exception if any.
+     */
+    public void commitRecord(String logLine) throws IOException {
+        commitRecords(Arrays.asList(logLine));
+    }
+
+    /**
+     * Commits a given list of system log records to the underlying log chunk.
+     *
+     * @param records List of records to log to.
+     * @throws IOException Exception in case of any error.
+     */
+    synchronized public void commitRecords(Collection<String> records) throws IOException {
```

Comment:
Why is it necessary to synchronize and `LOCK`?

Reply:
Thanks! I think I forgot to take synchronized out. 
I'll try to reduce scope all locks in that file.

Reply:
fixed. Thanks